src/databricks/labs/ucx/assessment/dashboards.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+import itertools
+import json
+import logging
+from collections.abc import Iterable, Iterator
+from dataclasses import dataclass, field
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.labs.lsql.lakeview import Dashboard as LsqlLakeviewDashboard, Dataset
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import DatabricksError
+from databricks.sdk.service.dashboards import Dashboard as SdkLakeviewDashboard
+from databricks.sdk.service.sql import Dashboard as SdkRedashDashboard, LegacyQuery
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Query:
+    """UCX representation of a Query.
+
+    Note:
+        This class is not persisted into an inventory table. If you decide to persist this class, consider (future)
+        differences between Redash and Lakeview queries
+    """
+
+    id: str
+    """The ID for this query."""
+
+    name: str = "UNKNOWN"
+    """The title of this query that appears in list views, widget headings, and on the query page."""
+
+    parent: str = "ORPHAN"
+    """The identifier of the workspace folder containing the object."""
+
+    query: str = ""
+    """The text of the query to be run."""
+
+    @classmethod
+    def from_legacy_query(cls, query: LegacyQuery) -> Query:
+        """Create query from a :class:LegacyQuery"""
+        assert query.id
+        return cls(
+            id=query.id,
+            name=query.name or cls.name,
+            parent=query.parent or cls.parent,
+            query=query.query or cls.query,
+        )
+
+    @classmethod
+    def from_lakeview_dataset(cls, dataset: Dataset, *, parent: str | None = None) -> Query:
+        """Create query from a :class:Dataset"""
+        return cls(
+            id=dataset.name,
+            name=dataset.display_name or cls.name,
+            parent=parent or cls.parent,
+            query=dataset.query,
+        )
+
+
+@dataclass
+class RedashDashboard:
```

Comment:
I might merge this with the LakeviewDashboard object, only difference are the tags